pyomo/contrib/mindtpy/mip_solve.py

```diff
@@ -146,32 +158,45 @@ def handle_master_mip_optimal(master_mip, solve_data, config):
         contains the specific configurations for the algorithm
     """
     # proceed. Just need integer values
-    MindtPy = master_mip.MindtPy_utils
-    main_objective = next(
-        master_mip.component_data_objects(Objective, active=True))
+    if config.strategy in {'OA'}:
+        MindtPy = master_mip.MindtPy_utils
+        main_objective = next(
+            master_mip.component_data_objects(Objective, active=True))
+    elif config.strategy == 'feas_pump':
+        main_objective = next(solve_data.mip.component_data_objects(Objective, active=True))
+
     # check if the value of binary variable is valid
-    for var in MindtPy.variable_list:
+    for var in master_mip.MindtPy_utils.variable_list:
         if var.value is None and var.is_integer():
             config.logger.warning(
                 "Integer variable {} not initialized. It is set to it's lower bound when using the initial_binary initialization method".format(var.name))
             var.value = var.lb  # nlp_var.bounds[0]
+    
     # warm start for the nlp subproblem
-    copy_var_list_values(
-        master_mip.MindtPy_utils.variable_list,
-        solve_data.working_model.MindtPy_utils.variable_list,
-        config)
+    if config.strategy == 'OA':
+        copy_var_list_values(
+            master_mip.MindtPy_utils.variable_list,
+            solve_data.working_model.MindtPy_utils.variable_list,
+            config)
+    elif config.strategy == 'feas_pump':
+        copy_var_list_values(
+            master_mip.MindtPy_utils.variable_list,
+            solve_data.mip.MindtPy_utils.variable_list,
```

Comment:
I think master_mip and solve_data.mip are the same here. Is this the correct way to warm start?